packages/flow/lib/AstroIterate.astro.d.ts

```diff
@@ -1,17 +1,16 @@
-export default function For<T>(props: Props<T>): any
+import { GetAppropriateFunctionBasedOnWhetherOrNotAGeneratorOfAnIterableWithTheForEachMethodIsPassed } from "./shared"
 
-export type Props<T> = {
-	[K in keyof T]: K extends 'children' ? unknown : T[K]
-} & {
-	children?: {
-		(item: IterableValue<T[Exclude<keyof T, 'children'>]>): any
-	}
+export default function For<T extends Iterable<unknown>>(props: Props<T>): any
+
+export type Props<T extends Iterable<unknown>> =  {
+	of: T
```

Comment:
This is the right type for the `<For/>` component.  The function that is passed in is done based on the iterable that is passed in. If the iterable has a `forEach()` method then the function will pass in the value, key or index , and the iterable to the callback function If not the value will be passed